perf/evp_fetch.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <openssl/evp.h>
+#include <openssl/kdf.h>
+#include <openssl/core_names.h>
+#include "perflib/perflib.h"
+
+#define NUM_CALLS_PER_TEST         100000
+
+OSSL_TIME *times;
+
+int err = 0;
+
+static int threadcount;
+
+static OSSL_LIB_CTX *ctx = NULL;
+
+#define ARRAY_SIZE(a)                               \
+  ((sizeof(a) / sizeof(*(a))))
+
+typedef enum {
+    FETCH_MD = 0,
+    FETCH_CIPHER,
+    FETCH_KDF,
+    FETCH_MAC,
+    FETCH_RAND
+} fetch_type_t;
+
+struct fetch_data_entry {
+    fetch_type_t ftype;
+    const char *alg;
+    const char *propq;
+};
+
+static struct fetch_data_entry fetch_entries[] = {
```

Comment:
The entries in this table should be conditional based on the `OPENSSL_NO_xxx` defines from `include/openssl/configuration.h`.  Alternatively, only use things that are guaranteed to exist.